### PR TITLE
Fixing Council pages,

### PR DESCRIPTION
**Pull request by davidwessman**
Removes all mentions of the_role actions, fixing old controllers.

Album-, workpost-controllers are not in used at the moment and I will not implement tests for them now.

### CODE REVIEW

**Comment by davidwessman** [comment]
Fixing #148 and #149

**Comment by davidwessman** [comment]
Did not have time to fix tests for this - would be great if someone can do some integration tests over the PageModule and PostModule.

**Comment by davidwessman** [comment]
Should fix #145 and replace PR #146

**Comment by davidwessman** [comment]
Fixes #148, #149, #151.

Still need some tests and Post-controller could need some work.
This should be tested on Dev - but it has some good changes which would be great to publish.

**Comment by davidwessman** [comment]
@henrikssn 
We need to get this out soon.
I couldn't get deploy this to http://samson.fsektionen.se/projects/web/deploys/82, any clues?

**Comment by henrikssn** [comment]
Fixat 974833b

**Comment by henrikssn** [comment]
There seems to be another problem with the translations. I'll leave that one to you.

**Comment by davidwessman** [comment]
@henrikssn 
Gonna try it out on dev, implemented test for #157  and fixed it.
Anything else?
(The function to give people a posts might not work correctly yet, but I started working on too much.)

**Comment by davidwessman** [comment]
@henrikssn 
http://dev.fsektionen.se this error feels deploy-related, any clues?

File: config/routes.rb [hunk 1]
```diff
@@ -126,29 +125,30 @@
...
-      resources :nominations, path: :nominera, only: [:create] do
-        get '', action: :new, on: :collection, as: :new
+    resources :elections, path: :val, only: :index do
+      collection do
+        resources :nominations, controller: 'elections/nominations',
+                  path: :nominera, only: [:create] do
+          get '', action: :new, on: :collection, as: :new
+        end
+        resources :candidates, controller: 'elections/candidates',
+                  path: :kandidera, except: [:edit]
```
Comment (davidwessman) [inline config/routes.rb]: @fsek/admins 
How is this supposed to look?

**Comment by SirDirac** [comment]
This PR was deployed to Production. Reference: <a href='http://samson.fsektionen.se/projects/web/deploys/88' target='_blank'>83f5b59</a>

**Comment by davidwessman** [comment]
The page link on the edit page doesn't work.
I think that is because I changed the page controller. Would need to fix this in the console.

File: config/routes.rb (hunk 1 above)
Comment (jforberg) [inline config/routes.rb]: I think the hound is wrong in this case. Turn off the flag if you can find it.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,102 +1,74 @@
...
-      @posts = Post.all
-    end
+    @posts = (@council.present?) ? @council.posts : Post.all
     @post_grid = initialize_grid(@posts)
   end
 
   def new
     @post = @council.posts.build
-    @councils = Council.order(title: :asc)
+    @post_permissions = @post.permissions.collect! { |p| p.id }
```
Comment (jforberg) [inline app/controllers/posts_controller.rb]: Yes. Also don't use the bang methods unless you have to. Even then, don't use the bang methods.

They modify their argument, after this line, @post.permissions will be a list of ints, not of permissions. Probably not what you want and in any case it's a code smell.

Comment (davidwessman) [inline app/controllers/posts_controller.rb]: @henrikssn Wrote this line, I'm not sure how it should be.

Comment (jforberg) [inline app/controllers/posts_controller.rb]: 1. Use ``permissions = @post.permissions.map &:id

2. No need to do this in the controller, do it when you need the values instead (in the view). Also then you don't have to put it in an instance variable. In general try to set only a single instance variable in the controller.

**Pull request merged**